Source/Relativity.Test.Helpers/Relativity.Test.Helpers/Kepler/OAuth2Helper.cs

```diff
@@ -0,0 +1,261 @@
+				if (oAuth.Exists(x => x.Name.Equals(oAuth2Name)))
+				{
+					if (oAuth.Exists(x => x.ContextUser == user.ArtifactID))
+					{
+						Console.WriteLine($"User ({user.EmailAddress}) already has OAuth2 Credentials within ({oAuth2Name})");
+
+						oAuth2Client = oAuth.Find(x => x.Name.Equals(oAuth2Name) && x.ContextUser == user.ArtifactID);
+					}
+					else
+					{
+						throw new Exception($"A different user was has OAuth2 Credentials within ({oAuth2Name})");
+					}
+				}
+				else
+				{
+					Services.Security.Models.OAuth2Client oAuthResult = await _oAuth2ClientManager.CreateAsync(oAuth2Name, OAuth2Flow.ClientCredentials, new List<Uri>(), user.ArtifactID);
+					if (oAuthResult != null)
+					{
+						Console.WriteLine($"Created OAuth2 Client Credentials for ({user.EmailAddress}) within ({oAuthResult.Name})");
+						oAuth2Client = oAuthResult;
+					}
+					else
+					{
+						throw new Exception($"Failed to create OAuth2 Client Credentials for ({user.EmailAddress}) within ({oAuth2Name})");
+					}
+				}
+
+				return oAuth2Client;
+			}
+			catch (Exception ex)
+			{
+				throw new Exception($"Exception occurred in ({nameof(CreateOAuth2ClientAsync)})", ex);
+			}
+		}
+
+		/// <summary>
+		/// Use this to get the Bearer Token that will give you access Relativity and various endpoints
+		/// </summary>
+		/// <param name="protocol"></param>
+		/// <param name="serverAddress"></param>
+		/// <param name="clientId"></param>
+		/// <param name="clientSecret"></param>
+		/// <param name="scope"></param>
+		/// <param name="grantType"></param>
+		/// <param name="httpMessageHandler"></param>
+		/// <returns></returns>
+		public async Task<string> GetBearerTokenAsync(string protocol, string serverAddress, string clientId, string clientSecret, string scope = "SystemUserInfo", string grantType = "client_credentials", HttpMessageHandler httpMessageHandler = null)
+		{
+			try
+			{
+				string tokenUrl = GetTokenUrl(protocol, serverAddress);
+
+				if (httpMessageHandler == null)
+				{
+					httpMessageHandler = new HttpClientHandler()
+					{
+						AllowAutoRedirect = false,
+						UseDefaultCredentials = true
+					};
+				}
+
+				string tokenResponse = await GetTokenAsync(tokenUrl, clientId, clientSecret, scope, grantType, httpMessageHandler);
+
+				dynamic data = JObject.Parse(tokenResponse);
+				string bearerToken = data.access_token.Value;
+
+				return bearerToken;
+			}
+			catch (Exception ex)
+			{
+				throw new Exception($"Exception occurred in ({nameof(GetBearerTokenAsync)})", ex);
+			}
+		}
+
+		/// <summary>
+		/// Use this to cleanup OAuth2 after tests
+		/// </summary>
+		/// <param name="clientId"></param>
+		/// <returns></returns>
+		public async Task DeleteOAuth2ClientAsync(string clientId)
+		{
+			try
+			{
+				await _oAuth2ClientManager.DeleteAsync(clientId);
+			}
+			catch (Exception ex)
+			{
+				throw new Exception($"Exception occurred in ({nameof(DeleteOAuth2ClientAsync)})", ex);

```

Comment:
are we still throwing system exceptions?

Reply:
I fixed these in the latest branch, you will see the changes there.